borg/repository.py

```diff
@@ -308,6 +317,8 @@ def complete_xfer():
         self.compact = set()
 
     def replay_segments(self, index_transaction_id, segments_transaction_id):
+        # fake an old client, so that in case we do not have an exclusive lock yet, prepare_txn will upgrade the lock:
+        self.exclusive = None
```

Comment:
prepare_txn will upgrade the lock.
either it will succeed, then we have an exclusive lock or it will fail with an exception.

so, for cleanliness we could remember and reset to previous value, but in practice I guess it makes no difference.